pyproject.toml

```diff
@@ -1,3 +1,97 @@
+[project]
+name = "frappe"
+authors = [
+    { name = "Frappe Technologies Pvt Ltd", email = "developers@frappe.io"}
+]
+description = "Metadata driven, full-stack low code web framework"
+requires-python = ">=3.8"
+readme = "README.md"
+dynamic = ["version"]
+dependencies = [
```

Comment:
Update CI configs to hash pyproject to for cache validation

Reply:
dev dependencies should ideally also move here 👀

Reply:
> dev dependencies should ideally also move here :eyes:

Ideally. Maybe it would look something like https://github.com/frappe/frappe/issues/13811#issuecomment-1155153968

Reply:
We'll have to defer getting rid of dev-dependencies.txt further until bench v5.12.0 gets widely adopted...and if we're going to miss the v14 release then wait until v15 to do so 🤔

Reply:
BTW I doubt many users have their own dev-requirements 😬 (scanned on source graph)

We can fix frappe apps and deprecate this totally too. It was never released (v14 feature)